Mlem/App/Utility/Extensions/Content Models/Comment1Providing+Extensions.swift

```diff
@@ -188,4 +193,40 @@ extension Comment1Providing {
             callback: callback
         )
     }
+    
+    func collapseAction(commentTreeTracker: CommentTreeTracker? = nil) -> BasicAction {
+        .init(
+            id: "collapse\(uid)",
+            appearance: .collapse(),
+            callback: { @MainActor in
+                withAnimation(UIAccessibility.isReduceMotionEnabled ? nil : .default) {
+                    commentTreeTracker?.nodesKeyedByActorId[self.actorId]?.collapsed.toggle()
+                }
+            }
+        )
+    }
+    
+    func collapseParentAction(commentTreeTracker: CommentTreeTracker? = nil) -> BasicAction {
+        .init(
+            id: "collapseParent\(uid)",
+            appearance: .collapseParent(),
+            callback: { @MainActor in
+                withAnimation(UIAccessibility.isReduceMotionEnabled ? nil : .default) {
+                    commentTreeTracker?.nodesKeyedByActorId[self.actorId]?.parent?.collapsed.toggle()
```

Comment:
I wonder if this should coalesce to the own comment if there's no parent 🤔 Something like:

```suggestion
                    guard let comment = commentTreeTracker?.nodesKeyedByActorId[self.actorId] else { return }
                    (comment?.parent ?? comment).collapsed.toggle()
```

This already works for the "collapse to top" option, so no changes are needed there.